js/components/ScriptChart.js

```diff
@@ -1,13 +1,214 @@
 import React from "react";
 
-import DragAndDropTable from "./DragAndDropTable";
+/* ScriptChart - This is the central script chart/table rendering
+ * component, situated between its parent DashTabs component, which
+ * stores most of the state of the scriptchart contents, and the
+ * DragTable component, which implements lower-level functions of
+ * the scriptchart.
+ * It instantiates the rows and columns via props passed down from
+ * submission of the selection/configuration form (columnManuscripts,
+ * rowLetters), and populates them -- filling in the rows -- 
+ * with data queried from the API.
+ * It also passes up requests to move and hide rows and columns
+ * and to display manuscripts in the viewer, and is re-rendered
+ * when these changes are processed.
+ */
+
+import DragTable from "./DragTable";
+import LetterImage from "./LetterImage";
+import ColumnControls from "./ColumnControls";
+import SyriacLetter from "./SyriacLetter";
+
+import "./ScriptChart.css";
 
 class ScriptChart extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.getRows = this.getRows.bind(this);
+    this.getColumns = this.getColumns.bind(this);
+    this.onRow = this.onRow.bind(this);
+    this.onHideColumn = this.onHideColumn.bind(this);
+    this.onHideRow = this.onHideRow.bind(this);
+    this.viewManifest = this.viewManifest.bind(this);
+  }
+
+  /* Table data is stored as a two-dimensional array.
+   * First key: letter ID (row)
+   * Second key: manuscript ID (column)
+   * Value: an Object containing the data necessary to render
+   * 1-5 image letter excerpts (binarized or not). This includes
+   * the link to the page URL and the coordinates of each letter.
+   */
+
+  viewManifest(manifestURL, manifestActivator) {
+    manifestActivator(manifestURL);
+  }
+
+  onHideColumn(manuscriptID, columnHider) {
+    columnHider(manuscriptID);
+  }
+
+  getRows() {
+    let rows = [];
+
+    /* Add row that will contain link to Mirador viewer, column hider Xs */
+    let colControls = { id: 0, ltid: "", letter: "", visible: true };
+
+    /* Add dates row */
+    let datesRow = {
+      id: 1,
+      ltid: "Date",
+      letter: "Date",
+      ltid: "Date",
```

Comment:
It seems like "ltid" is in this object twice as a key. Is that necessary?

Reply:
Oops -- definitely not necessary.